source/game.c

```diff
@@ -2203,287 +2208,351 @@ static bool check_and_score_joker_for_event(ListItr* starting_joker_itr, CardObj
+{
+    if (played_idx == 0 && (timer % FRAMES(30) == 0) && timer > FRAMES(40))
+    {
 
-                // Score Jokers normally
-                case PLAY_SCORING_JOKERS:
+        tte_erase_rect_wrapper(PLAYED_CARDS_SCORES_RECT);
 
-                    if (i == 0 && (timer % FRAMES(30) == 0) && timer > FRAMES(40))
-                    {
+        if (check_and_score_joker_for_event(&_joker_scored_itr, NULL, JOKER_EVENT_INDEPENDENT))
+        {
+            return true;
+        }
 
-                        tte_erase_rect_wrapper(PLAYED_CARDS_SCORES_RECT);
+        // Trigger hand end effect for all jokers once they are done scoring
+        if (check_and_score_joker_for_event(&_joker_round_end_itr, NULL, JOKER_EVENT_ON_HAND_SCORED_END))
+        {
+            return true;
+        }
 
-                        if (check_and_score_joker_for_event(&_joker_scored_itr, NULL, JOKER_EVENT_INDEPENDENT))
-                        {
-                            return;
-                        }
+        play_state = PLAY_ENDING;
+        timer = TM_ZERO;
+        scored_card_index = played_top + 1; // Reset the scored card index to the top of the played stack
+    }
 
-                        // Trigger hand end effect for all jokers once they are done scoring
-                        if (check_and_score_joker_for_event(&_joker_round_end_itr, NULL, JOKER_EVENT_ON_HAND_SCORED_END))
-                        {
-                            return;
-                        }
+    return false;
+}
 
-                        play_state = PLAY_ENDING;
-                        timer = TM_ZERO;
-                        scored_card_index = played_top + 1; // Reset the scored card index to the top of the played stack
-                        break;
-                    }
-                
-                    if (card_object_is_selected(played[i]))
-                    {
-                        played_y -= int2fx(10);
-                    }
-                    break;
+// This is the reverse of PLAY_STARTING. The cards get reset back to their neutral position sequentially
+static void play_ending_update(int played_idx)
+{
+    if (played_idx == 0 && (timer % FRAMES(10) == 0 || !card_object_is_selected(played[played_top - scored_card_index])) && timer > FRAMES(40))
+    {
+        scored_card_index--;
 
-                case PLAY_ENDING: // This is the reverse of PLAY_PLAYING. The cards get reset back to their neutral position sequentially
+        if (scored_card_index == 0)
+        {
+            play_state = PLAY_ENDED;
+            timer = TM_ZERO;
+        }
+    }
 
-                    if (i == 0 && (timer % FRAMES(10) == 0 || !card_object_is_selected(played[played_top - scored_card_index])) && timer > FRAMES(40))
-                    {
-                        scored_card_index--;
+    if (card_object_is_selected(played[played_idx]))
+    {
+        played[played_idx]->sprite_object->ty = int2fx(70);
```

Comment:
```suggestion
        played[played_idx]->sprite_object->ty = int2fx(HAND_PLAY_POS.y);
```
Now that the constant is defined we can remove this magic number